include/execution.hpp

```diff
@@ -2171,7 +2172,257 @@ namespace std::execution {
   /////////////////////////////////////////////////////////////////////////////
   // [execution.senders.adaptors.split]
   namespace __split {
+
+    struct __operation_base {
+      virtual void __notify() noexcept = 0;
+      virtual ~__operation_base() = default;
+    };
+
+    template <class _Value, class _Error>
+      class __receiver {
+        __operation_base &__op_state_;
+        in_place_stop_source &__stop_source_;
+        variant<_Value, _Error, set_stopped_t> &__data_;
+
+      public:
+        template <class... _As>
+        friend void tag_invoke(set_value_t, __receiver&& __self, _As&&... __as) noexcept {
+          try {
+            __self.__data_.template emplace<0>(tuple<decay_t<_As>...>{(_As &&) __as...});
+          } catch (...) {
+            __self.__data_.template emplace<1>(current_exception());
+          }
+          __self.__op_state_.__notify();
+        }
+
+        template <class _T>
+        friend void tag_invoke(set_error_t, __receiver&& __self, _T&& __error) noexcept {
+          __self.__data_.template emplace<1>(__error);
+          __self.__op_state_.__notify();
+        }
+
+        friend void tag_invoke(set_stopped_t tag, __receiver&& __self) noexcept {
+          ((__receiver&&)__self).__data_.template emplace<2>(tag);
+          __self.__op_state_.__notify();
+        }
+
+        friend auto tag_invoke(get_env_t, const __receiver& __self) {
+          return make_env<get_stop_token_t>(__self.__stop_source_.get_token());
+        }
+
+        __receiver(__operation_base &__op_state,
+                   in_place_stop_source &__stop_source,
+                   variant<_Value, _Error, set_stopped_t> &__data) noexcept
+          : __op_state_(__op_state)
+          , __stop_source_(__stop_source)
+          , __data_(__data) {
+        }
+      };
+
+    enum class __state_t { __created, __started, __completed };
+
+    template <class _Sender>
+      struct __sh_state : __operation_base {
+        using __nullable_variant_t = __bind_front<__munique<__q<variant>>, tuple<>>;
+        using __error = __error_types_of_t<_Sender,
+                                           __empty_env,
+                                           __bind_front<__q<__variant>,
+                                                        exception_ptr>>;
+        using __value = __value_types_of_t<_Sender,
+                                           __empty_env,
+                                           __q<__decayed_tuple>,
+                                           __nullable_variant_t>;
+        using __receiver = __receiver<__value, __error>;
+
+        in_place_stop_source __stop_source_{};
```

Comment:
I think this needs to be (and according to P2300R4 this is) `std::stop_source`, not `std::in_place_stop_source`. I think the cancellation scopes of the different `split()` senders are non-overlapping. Registration and unregistration of `in_place_stop_callback`s assume the cancellation scopes are strictly nested.

@lewissbaker can correct me if I'm wrong.